Error compiling locally: cannot find GOROOT directory: /usr/local/go

PR #37665 has broken my local build (`make`, `hack/local-up-cluster.sh`). It unsets GOROOT and go build then looks into `/usr/local/go` which does not exist.

```
$ echo $GOROOT
/home/jsafrane/.gimme/versions/go1.7.linux.amd64/

$ make
+++ [1201 11:13:33] Generating bindata:
    /home/jsafrane/project/go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1201 11:13:33] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1201 11:13:33] Building go targets for linux/amd64:
    cmd/libs/go2idl/deepcopy-gen
go: cannot find GOROOT directory: /usr/local/go
go: cannot find GOROOT directory: /usr/local/go
go: cannot find GOROOT directory: /usr/local/go
Makefile.generated_files:304: recipe for target '_output/bin/deepcopy-gen' failed
```

@ixdy, PTAL, it's really annoying regression.

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -219,8 +219,6 @@ kube::golang::set_platform_envs() {
   [[ -n ${1-} ]] || {
     kube::log::error_exit "!!! Internal error. No platform set in kube::golang::set_platform_envs"
   }
-  # make sure we have a clean slate first
-  kube::golang::unset_platform_envs
 
   export GOOS=${platform%/*}
   export GOARCH=${platform##*/}

